promlog/log.go

```diff
@@ -167,10 +169,15 @@ func (l *logger) SetLevel(lvl *AllowedLevel) {
 	l.mtx.Lock()
 	defer l.mtx.Unlock()
 	if lvl != nil {
-		if l.currentLevel != nil && l.currentLevel.s != lvl.s {
+		if (l.currentLevel != nil && l.currentLevel.s != lvl.s) || l.currentLevel == nil {
+			l.leveled = log.With(l.base, "ts", timestampFormat, "caller", log.Caller(5))
 			_ = l.base.Log("msg", "Log level changed", "prev", l.currentLevel, "current", lvl)
```

Comment:
If I understand the logic correctly, this message will now also be logged when a new dynamic logger is created. I don't think that's expected.